src/block/NetherVines.php

```diff
@@ -0,0 +1,178 @@
+
+	protected int $age = 0;
+
+	public function getRequiredStateDataBits() : int{
+		return 5;
+	}
+
+	public function describeState(RuntimeDataWriter|RuntimeDataReader $w) : void{
+		$w->boundedInt(5, 0, self::MAX_AGE, $this->age);
+	}
+
+	public function getAge() : int{
+		return $this->age;
+	}
+
+	/** @return $this */
+	public function setAge(int $age) : self{
+		if($age < 0 || $age > self::MAX_AGE){
+			throw new \InvalidArgumentException("Age must be in range 0-" . self::MAX_AGE);
+		}
+
+		$this->age = $age;
+		return $this;
+	}
+
+	public function isAffectedBySilkTouch() : bool{
+		return true;
+	}
+
+	public function ticksRandomly() : bool{
+		return true;
+	}
+
+	public function canClimb() : bool{
+		return true;
+	}
+
+	abstract protected function getGrowthFace() : int;
+
+	protected function getSupportFace() : int{
+		return Facing::opposite($this->getGrowthFace());
+	}
+
+	private function canBeSupportedBy(Block $block) : bool{
+		return $block->getSupportType($this->getSupportFace())->hasCenterSupport() || $block->isSameType($this);
+	}
+
+	public function onNearbyBlockChange() : void{
+		if(!$this->canBeSupportedBy($this->getSide($this->getSupportFace()))){
+			$this->position->getWorld()->useBreakOn($this->position);
+		}
+	}
+
+	private function seekToTop() : NetherVines{
+		$face = $this->getGrowthFace();
+		$top = $this;
+		while(($next = $top->getSide($face))->isSameType($this)){
+			$top = $next;
+		}
+		return $top;
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if(!$this->canBeSupportedBy(($support = $blockReplace->getSide($this->getSupportFace())))){
+			return false;
+		}
+		$this->age = $support->isSameType($this) ? min($support->getAge() + 1, self::MAX_AGE) : mt_rand(0, self::MAX_AGE - 1);
+		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($item instanceof Fertilizer){
+			if($this->seekToTop()->getSide($this->getGrowthFace())->canBeReplaced()){
+				if($this->grow($player, mt_rand(1, 5))){
+					$item->pop();
+				}
+			}
+			return true;
+		}
+		return false;
+	}
+
+	public function onRandomTick() : void{
+		if(mt_rand(0, 100) <= 10 && $this->age < self::MAX_AGE){
+			if($this->seekToTop()->getSide($this->getGrowthFace())->canBeReplaced()){
+				$this->grow(null, 1);
+			}
+		}
+	}
+
+	private function grow(?Player $player, int $growthAmount) : bool{
+		$top = $this->seekToTop();
+		$face = $this->getGrowthFace();
+
+		$tx = new BlockTransaction($this->position->getWorld());
+		for($i = 1; $i <= $growthAmount && $top->getSide($face, $i)->canBeReplaced(); $i++){
+			$tx->addBlock($top->getSide($face, $i)->getPosition(), (clone $top)->setAge(min($top->getAge() + $i, self::MAX_AGE)));
```

Comment:
This won't update the block itself, so the age of the block will stay the same.

Reply:
What do you mean? The age of the block is updating

Video: https://www.youtube.com/watch?v=W74C13sN_Kk

Reply:
I don't think you're understanding me. The age of the bottom block (where the growth originated from) isn't being updated.